### DISCUSSION TITLE
What should be the endpoint address when I use Flutter + Deno + MongoDB?

### DISCUSSION THREAD

**Discussion by goodmanworld** [denoland/deno#12580] [Q&A]
I am using Flutter, VSCode and Android emulator in Windows 10. Also Deno as the backend server. I am trying to connect Flutter app to Deno app locally in my PC. But I don't know what address exactly I must use?

This is the address I am trying to send `get, post, etc.` requests from the Flutter app to:

    http://10.0.2.2:8000/api/register

And this is the address I am trying to receive those requests inside my deno app:

    Listening on: http://0.0.0.0:8000

And the endpoints such as `/api/register` work when I use postman to test them with the follwong address for example:

    0.0.0.0:8000/api/register

I don't know what is the problem and how can I fix it?

**Comment by lucacasonato** [comment]
This is more of an Android emulator issue, rather than a Deno issue. I suggest you ask around on the Deno Discord (https://discord.gg/deno), or in the Flutter community.

**Comment by Soremwar** [comment]
Why don't you try to make the Deno backend listen on you public IP directly?

0.0.0.0 means you are gonna listen on all IPs available on your machine, however if 10.0.0.2 is not configured as your public IP you won't be able to listen for anything there

**Reply by goodmanworld** [reply]
> Why don't you try to make the Deno backend listen on you public IP directly?
> 
> 0.0.0.0 means you are gonna listen on all IPs available on your machine, however if 10.0.0.2 is not configured as your public IP you won't be able to listen for anything there

I think my Deno app listens to `0.0.0.0:8000` address and it works when I try with the Postman. But when I try to send my requests from Flutter to that address, I get the following error message:

`socketexception os error connection refused errno = 111 address = 0.0.0.0.port=33668`

**Reply by Soremwar** [reply]
That error is strange, Flutter says it's trying to connect to the port 33668 instead of 8000. You should probably check the line where you try to connect to that endpoint
